### ISSUE TITLE
The count of user assigned identity specified in identity.userAssignedIdentities is 2, while expected value is 1.

### ISSUE THREAD

**Issue by Chivolta** [hashicorp/terraform-provider-azurerm#16890]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.0

### AzureRM Provider Version

3.7.0

### Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
dynamic "identity" {
    for_each = var.cluster_identity != null ? [true] : []
    content {
      type = "UserAssigned"
      identity_ids = [
        data.azurerm_user_assigned_identity.cluster[0].id,
        var.additional_identity_ids[0],
      ]
    }
  }
```

### Debug Output/Panic Output

```shell
│ Error: updating Cluster: (Managed Cluster Name "...." / Resource Group "...."): containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest" Message="The count of user assigned identity specified in identity.userAssignedIdentities is 2, while expected value is 1."
│ 
│   with module.aks.azurerm_kubernetes_cluster.this,
│   on .terraform/modules/aks/main.tf line 84, in resource "azurerm_kubernetes_cluster" "this":
│   84: resource "azurerm_kubernetes_cluster" "this" {
│ 
╵
##[error]Terraform command 'apply' failed with exit code '1'.
##[error]╷
│ Error: updating Cluster: (Managed Cluster Name "..." / Resource Group "..."): containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest" Message="The count of user assigned identity specified in identity.userAssignedIdentities is 2, while expected value is 1."
│ 
│   with module.aks.azurerm_kubernetes_cluster.this,
│   on .terraform/modules/aks/main.tf line 84, in resource "azurerm_kubernetes_cluster" "this":
│   84: resource "azurerm_kubernetes_cluster" "this" {
│ 
╵
```

### Expected Behaviour

Two user assigned identities should be assigned.

### Actual Behaviour

Terraform plan works correctly:
```
      ~ identity {
          ~ identity_ids = [
              + "/subscriptions/.../resourceGroups/.../providers/Microsoft.ManagedIdentity/userAssignedIdentities/...",
                # (1 unchanged element hidden)
            ]
            # (1 unchanged attribute hidden)
        }
```

Terraform apply throws error:
`Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest" Message="The count of user assigned identity specified in identity.userAssignedIdentities is 2, while expected value is 1."`

### Steps to Reproduce

1. terraform plan
2. terraform apply

**Comment by ms-henglu** [comment]
Hi @Chivolta ,

I just confirmed with service team, and this is by design to support 1 user assigned identity, and currently there's no plan to support more.

**Comment by stephybun** [comment]
Closed by #17299
